packages/app/components/stackbox/Stackbox.tsx

```diff
@@ -60,39 +60,67 @@ const balanceOptions = [
 
 export const Stackbox = () => {
   const searchTokenBarRef = useRef<HTMLInputElement>(null);
-  const [isPickingTokenFrom, setIsPickingTokenFrom] = useState<boolean>(false);
-  const [tokenFrom, setTokenFrom] = useState<TokenFromTokenlist>();
-  const [tokenTo, setTokenTo] = useState<TokenFromTokenlist>();
+  const [isPickingFromToken, setIsPickingFromToken] = useState<boolean>(false);
+  const [fromToken, setFromToken] = useState<TokenFromTokenlist>();
+  const [toToken, setToToken] = useState<TokenFromTokenlist>();
   const { closeModal, openModalId, openModal } = useModalContext();
 
   const { chain } = useNetwork();
   const { address } = useAccount();
   const { data: balance } = useBalance({
-    address: Boolean(tokenFrom) ? address : undefined,
-    token: tokenFrom?.address as `0x${string}`,
+    address: Boolean(fromToken) ? address : undefined,
+    token: fromToken?.address as `0x${string}`,
     chainId: chain?.id,
   });
   const [tokenAmount, setTokenAmount] = useState("");
 
-  const [frequency, setFrequency] = useState<string>(HOUR_OPTION);
+  const [frequency, setFrequency] = useState<FREQUENCY_OPTIONS>(
+    FREQUENCY_OPTIONS.hour
+  );
   const [startDateTime, setStartDateTime] = useState<Date>(
     new Date(startDateTimeTimestamp)
   );
   const [endDateTime, setEndDateTime] = useState<Date>(
     new Date(endDateByFrequency[frequency])
   );
 
-  const openConfirmStack = () => openModal(ModalId.CONFIRM_STACK);
-  const openTokenPicker = (isTokenFrom = true) => {
-    setIsPickingTokenFrom(isTokenFrom);
-    openModal(ModalId.TOKEN_PICKER);
-  };
-  const selectToken = isPickingTokenFrom ? setTokenFrom : setTokenTo;
+  const [showTokenAmountError, setShowTokenAmountError] = useState(false);
+  const [showDateTimeError, setShowDateTimeError] = useState(false);
+  const [showFromTokenError, setShowFromTokenError] = useState(false);
+  const [showToTokenError, setShowToTokenError] = useState(false);
```

Comment:
WDYT about dropping these `useState` variables and storing the error styles in a variable for readability purposes? Feels easier than toggling on and off checking other state variables.

Maybe I'm missing something tho.

Reply:
What do you mean with error styles in a variable? I can only see `useReducer` simplifying this.

Reply:
Oh, you mean having a style directly defined in the variable instead of a bool value. That also can work. But can't use a `useReducer` on it. Don't know which path is better.

Reply:
Which reducer are you talking about? D: 
I can't find it.

On my suggestion about the styles into a variable, it's because I thought you created these `useState` variables for mostly conditional rendering and/or conditional styling (to keep conditional styles names meaningfully).

What I see is:
`showTokenAmountError` -> conditional styling
`showDateTimeError` -> conditional rendering
`showFromTokenError` & `showToTokenError` -> conditional styling + animation event trigger

The only place where these values go `true` is when the `Stack now` button gets clicked but all the 5 values (the 2 dates and the token values) are either coming from a hook or stateful variables.

So my doubt is around why do we need these 4 error stateful variables if we could simply use their respective dependency values.

But maybe I'm missing something like the reducer you're talking about but I can't find it D:

Reply:
I'm talking about `useReducer` as a way to improve this code :)

Reply:
I think we can had a `useReducer` in the future. It will make the component cleaner. For now went with you suggestion! Tell me what you think.

Reply:
Ohhh I get now what you said. Yeah, I'm ok with that.

I saw the discussion on the changes, it's my bad for poorly explaining what I had in mind, next time I'll bring a block of code as a suggestion with the explanation, sorry for the extra work :c